environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/ResourceEncryptionInitializationHandler.java

```diff
@@ -78,6 +116,47 @@ public void accept(Event<EnvironmentDto> environmentDtoEvent) {
         }
     }
 
+    private Optional<String> getAzureResourceGroupName(EnvironmentDto environmentDto, Environment environment) {
+        Optional<String> emptyResponse = Optional.empty();
+        if (isSingleResourceGroup(environmentDto)) {
+            return Optional.of(environmentDto)
+                    .map(EnvironmentDto::getParameters)
+                    .map(ParametersDto::getAzureParametersDto)
+                    .map(AzureParametersDto::getAzureResourceGroupDto)
+                    .map(AzureResourceGroupDto::getName);
+        }
+        return emptyResponse;
+    }
+
+    private boolean isSingleResourceGroup(EnvironmentDto environmentDto) {
+        ResourceGroupUsagePattern resourceGroupUsagePattern = Optional.ofNullable(environmentDto.getParameters())
+                .map(ParametersDto::azureParametersDto)
+                .map(AzureParametersDto::getAzureResourceGroupDto)
+                .map(AzureResourceGroupDto::getResourceGroupUsagePattern)
+                .orElse(USE_MULTIPLE);
+        return resourceGroupUsagePattern.isSingleResourceGroup();
+    }
+
+    private Object getOrCreateResourceGroup(EnvironmentDto environmentDto, Environment environment) {
+        Optional<String> resourceGroupName = getAzureResourceGroupName(environmentDto, environment);
+        if (resourceGroupName.isPresent()) {
+            return resourceGroupName;
+        } else {
+            CloudCredential cloudCredential = credentialToCloudCredentialConverter.convert(environment.getCredential());
+            AzureClient azureClient = azureClientService.getClient(cloudCredential);
+            String region = environment.getRegionSet().stream().findFirst().get().getName();
+            Map<String, String> tags = Collections.unmodifiableMap(environment.getEnvironmentTags().getUserDefinedTags());
+            ResourceGroup resourceGroup;
+            String resourceGroupNameForCreation = azureUtils.generateResourceGroupNameByNameAndId(
+                    String.format("%s-", environment.getName()),
+                    UUID.randomUUID().toString());
+            LOGGER.debug("Creating resource group {}", resourceGroupNameForCreation);
+            resourceGroup = azureClient.createResourceGroup(resourceGroupNameForCreation, region, tags);
```

Comment:
The RG name generation logic shall be moved over to `cloud-azure`, presumably into `AzureEncryptionResources` or into some new service invoked therein. The `environment.getName()` thing is the "base name" I mentioned above.

By the way, you will have to generate the DES name in a similar way, using the base name, adding some suffix (e.g. `-des`) then appending the UUID like for the RG name here.